components/script/dom/htmlselectelement.rs

```diff
@@ -28,7 +32,8 @@ use style::element_state::*;
 
 #[dom_struct]
 pub struct HTMLSelectElement {
-    htmlelement: HTMLElement
+    htmlelement: HTMLElement,
+    options: MutNullableHeap<JS<HTMLOptionsCollection>>,
```

Comment:
It's expensive to create an HTMLFormControlsCollection for HTMLFormElement, but for HTMLSelectElement, creating an HTMLOptionsCollection is just a straightforward filter, hence it isn't necessary.